forge/src/executor/inspector/logs.rs

```diff
@@ -115,3 +85,46 @@ fn convert_hh_log_to_event(call: HardhatConsoleCalls) -> RawLog {
         data: ethers::abi::encode(&[Token::String(call.to_string())]),
     }
 }
+
+/// Extracts a log from the interpreter if there is any.
+pub fn extract_log(interpreter: &Interpreter) -> Option<RawLog> {
+    let num_topics = match interpreter.contract.code[interpreter.program_counter()] {
+        opcode::LOG0 => 0,
+        opcode::LOG1 => 1,
+        opcode::LOG2 => 2,
+        opcode::LOG3 => 3,
+        opcode::LOG4 => 4,
+        _ => return None,
+    };
+
+    let (offset, len) = (
+        as_usize_or_return!(interpreter.stack().peek(0).ok()?, None),
```

Comment:
this is fine, but can we try to keep macros to an absolute minimum? they are the worst for long term maintainability

Reply:
Yeah, most of the macros in this port are super tedious conversions (`as_usize_or_return!` for example) with the exception of the one used in `InspectorStack` that is just a convenience thing for static dispatch